File: DSA/Vyom-Singhal/Day13/Day13-q2.cpp
#include <iostream>
using namespace std;

int reverseNum(int n, int rev = 0) {
    if (n == 0) return rev;
    rev = rev * 10 + n % 10;
    return reverseNum(n / 10, rev);
}

string isPalindrome(int n) {
    int rev = reverseNum(n);
    if(n == rev) return "YES";
    return "NO";
}

int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;
    cout << isPalindrome(n);
}

File: DSA/Ishita-Nigam/Day13/Day13_Q2.java
import java.util.Scanner;

public class Day13_Q2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the number:");
        int number = scanner.nextInt();
        scanner.close();
        if (isPalindrome(number)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    public static boolean isPalindrome(int number) {
        String str = Integer.toString(number);
        return isPalindromeHelper(str, 0, str.length() - 1);
    }

    private static boolean isPalindromeHelper(String str, int start, int end) {
        if (start >= end) {
            return true;
        }

        if (str.charAt(start) != str.charAt(end)) {
            return false;
        }

        // Recursively check the next pair of characters
        return isPalindromeHelper(str, start + 1, end - 1);
    }
}
